src/gpu/sycl/ref_pooling.hpp

```diff
@@ -53,18 +53,13 @@ struct ref_pooling_fwd_t : public sycl_gpu_primitive_t {
 
             const bool ok = is_fwd() && set_default_params() == status::success
                     && (src_md(0)->format_desc.blocking.inner_nblks == 0)
-                    && (utils::everyone_is(
-                                s8, src_md(0)->data_type, dst_md(0)->data_type)
-                            || utils::everyone_is(u8, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(f32, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(bf16, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(f16, src_md(0)->data_type,
-                                    dst_md(0)->data_type)
-                            || utils::everyone_is(s32, src_md(0)->data_type,
-                                    dst_md(0)->data_type))
+                    && (!utils::one_of(f64, src_md(0)->data_type)
+                            && !utils::one_of(f64, dst_md(0)->data_type))
+                    && (IMPLICATION(utils::one_of(bf16, src_md(0)->data_type),
+                            utils::one_of(bf16, dst_md(0)->data_type)))
```

Comment:
Could you explain, please, what makes bf16 so special?

Reply:
In this instance, we're adhering to the specifications outlined in the table available [here](https://oneapi-src.github.io/oneDNN/dev_guide_pooling.html#data-type-support). Notably, the `bf16` scenario stands out as one case requiring identical source and destination data types. Thanks for the comment. Let me know if something else needs to be clarified.

Reply:
I don't see a strong reason to limit the support if it can work as expected.